waspc/packages/deploy/src/providers/fly/setup/setup.ts

```diff
@@ -0,0 +1,70 @@
+import { $, cd } from 'zx'
+import crypto from 'crypto'
+import * as tomlHelpers from '../helpers/tomlFileHelpers.js'
+import { DeploymentInfo, IDeploymentInfo } from '../DeploymentInfo.js'
+import { IGlobalOptions } from '../IGlobalOptions.js'
+import { buildDirExists, cdToClientBuildDir, cdToServerBuildDir, lazyInit, waspSays } from '../helpers/helpers.js'
+
+export async function setup(baseName: string, region: string, options: IGlobalOptions) {
+  waspSays(`Setting up your Wasp app with Fly.io!`)
+
+  const buildWaspIfMissing = lazyInit(async () => {
+    if (!buildDirExists(options.waspDir)) {
+      cd(options.waspDir)
+      await $`wasp build`
+    }
+  })
+
+  const tomlFiles = tomlHelpers.getTomlFileInfo(options)
+  const deploymentInfo = new DeploymentInfo(baseName, region, options, tomlFiles)
+
+  if (tomlHelpers.serverTomlExistsInProject(tomlFiles)) {
+    waspSays(`${tomlFiles.serverTomlPath} exists. Skipping server setup.`)
+  } else {
+    await buildWaspIfMissing()
+    await setupServer(deploymentInfo)
+  }
+
+  if (tomlHelpers.clientTomlExistsInProject(tomlFiles)) {
+    waspSays(`${tomlFiles.clientTomlPath} exists. Skipping client setup.`)
+  } else {
+    await buildWaspIfMissing()
+    await setupClient(deploymentInfo)
+  }
+
+  waspSays(`Don't forget to create your database by running the "create-db" command.`)
```

Comment:
Maybe we can offer to the user to run that command next with `y/N` kind of prompt? 

So the experience looks like:
1. User runs `setup`
2. `create-db` is chained right after it (with `y/N`)
3. `deploy` is chained next (with `y/N`)

If the user doesn't want to do that all at once, it's okay as well.

But this reads more like something for the next iteration 😄

Reply:
Ah, yeah this is a good idea. I can probably sneak it in. If not, the full commands in the output you suggested should help too, but I like the idea of just asking them if they want to run the next logical thing next. The only thing I wonder is perhaps this should be offered via some optional flag. So the base commands do just run once, but maybe we can add `--offer-next-command` or something to do what you describe. 🤔 hmm will think on it and see what others say, but I do like the idea of it.

Reply:
Ha a bit tricky yes!

Here is an idea:
We tell them to call `setup` command.
It calls all of these commands one by one -> setup, create-db, deploy.
If any of these steps fails, it tells you: I failed at this step, pls continue with the following command: `setup` or `setup db` or `deploy`.

So we would really have `setup`, `setup db` and `deploy`.

Although, maybe they already have database somewhere else and want to set up fly with using that database, is that possible? In that case we want to do sometihng different?

Maybe I am talking nonsense here @shayneczyzewski and it doesn't fit the use cases, so pls ignore it in that case!

Reply:
I think I like a mix of all the above. What if we kept the 3 existing commands, but we added one called `wasp deploy fly launch` (again stealing from fly), where that command just called these 3 in sequence? So all 3 behave as standalone (allowing like you said, maybe using an existing DB, which they could probably do using the UI or `wasp deploy fly cmd ...`), and we can check for errors after each, but we have one unique "shortcut" to chain them when they are just starting up their project?

Reply:
Sounds good!